modules/charon-core/src/main/java/org/wso2/charon3/core/utils/PatchOperationUtil.java

```diff
@@ -245,10 +245,46 @@ private static AbstractSCIMObject doPatchRemoveWithFiltersForLevelThree(Abstract
                         }
                     }
                 } else {
-                    throw new BadRequestException("Attribute : " + attributeParts[1] + " " +
-                            "is not a multivalued attribute.", ResponseCodeConstants.INVALID_PATH);
-                }
+                    // This is only valid for extensions.
+                    Attribute subSubAttribute = subAttribute.getSubAttribute(attributeParts[2]);
+                    if (subSubAttribute == null) {
+                        String error =
+                                String.format("No such sub attribute with the name : %s within the attribute %s",
+                                        attributeParts[2], attributeParts[1]);
+                        throw new BadRequestException(error, ResponseCodeConstants.INVALID_PATH);
+                    }
+                    if (!subSubAttribute.getMultiValued()) {
+                        String error = String.format("Attribute : %s is not a multivalued attribute.",
```

Comment:
We dont need to use String.format if we are only appending a single element

Reply:
addressed in https://github.com/wso2/charon/pull/286/commits/f738971766186c1ccf09b23ccb0cee39cacd016a